src/backend/distributed/executor/multi_utility.c

```diff
@@ -3770,3 +3776,54 @@ PlanGrantStmt(GrantStmt *grantStmt)
 
 	return ddlJobs;
 }
+
+
+/*
+ * ProcessDropTableStmt processes DROP TABLE commands for partitioned tables.
+ * If we are trying to DROP partitioned tables, we first need to go to MX nodes
+ * and DETACH partitions from their parents. Otherwise, we process DROP command
+ * multiple times in MX workers. For shards, we send DROP commands with IF EXISTS
+ * parameter which solves problem of processing same command multiple times.
+ * However, for distributed table itself, we directly remove related table from
+ * Postgres catalogs via performDeletion function, thus we need to be cautious
+ * about not processing same DROP command twice.
+ */
+static void
+ProcessDropTableStmt(DropStmt *dropTableStatement)
+{
+	ListCell *dropTableCell = NULL;
+
+	Assert(dropTableStatement->removeType == OBJECT_TABLE);
+
+	foreach(dropTableCell, dropTableStatement->objects)
+	{
+		List *tableNameList = (List *) lfirst(dropTableCell);
+		RangeVar *tableRangeVar = makeRangeVarFromNameList(tableNameList);
+		bool missingOK = true;
+		List *partitionList = NIL;
+		ListCell *partitionCell = NULL;
+
+		Oid relationId = RangeVarGetRelid(tableRangeVar, AccessExclusiveLock, missingOK);
+
+		if (relationId == InvalidOid ||
+			!IsDistributedTable(relationId) ||
```

Comment:
Minor: Can't we fit these 4 lines into 2 lines?

Reply:
We can, but I think this is more readable. If you insist I can change it.